ports/luasocket/CMakeLists.txt

```
@@ -22,33 +22,102 @@ add_library(socket.core
     src/select.c
     src/tcp.c
     src/udp.c
-    src/compat.c
-    src/wsocket.c)
-
-add_library(mime.core
-    src/mime.c
     src/compat.c)
-
+if(WIN32)
+    target_sources(socket.core PRIVATE
+        src/wsocket.c)
+elseif (UNIX)
+    target_sources(socket.core PRIVATE
+        src/usocket.c)
+endif()
+set_target_properties(socket.core PROPERTIES
+    RUNTIME_OUTPUT_DIRECTORY "socket"
+    LIBRARY_OUTPUT_DIRECTORY "socket"
+    PREFIX ""
+    RUNTIME_OUTPUT_NAME "core"
+    LIBRARY_OUTPUT_NAME "core")
 target_include_directories(socket.core PRIVATE ${LUASOCKET_INCLUDES})
 target_link_libraries(socket.core PRIVATE ${LUASOCKET_LIBRARIES})
 
+add_library(mime.core
+    src/mime.c
+    src/compat.c)
+set_target_properties(mime.core PROPERTIES
+    RUNTIME_OUTPUT_DIRECTORY "mime"
+    LIBRARY_OUTPUT_DIRECTORY "mime"
+    PREFIX ""
+    RUNTIME_OUTPUT_NAME "core"
+    LIBRARY_OUTPUT_NAME "core")
 target_include_directories(mime.core PRIVATE ${LUASOCKET_INCLUDES})
 target_link_libraries(mime.core PRIVATE ${LUASOCKET_LIBRARIES})
 
-add_definitions(
-    "-DLUASOCKET_API=__declspec(dllexport)"
-    "-DMIME_API=__declspec(dllexport)")
+if(UNIX)
+    add_library(socket.unix
+        src/buffer.c
+        src/compat.c
+        src/auxiliar.c
+        src/options.c
+        src/timeout.c
+        src/io.c
+        src/usocket.c
+        src/unix.c
+        src/unixdgram.c
+        src/unixstream.c)
+    set_target_properties(socket.unix PROPERTIES
+        RUNTIME_OUTPUT_DIRECTORY "socket"
+        LIBRARY_OUTPUT_DIRECTORY "socket"
+        PREFIX ""
+        RUNTIME_OUTPUT_NAME "unix"
+        LIBRARY_OUTPUT_NAME "unix")
+    target_include_directories(socket.unix PRIVATE ${LUASOCKET_INCLUDES})
+    target_link_libraries(socket.unix PRIVATE ${LUASOCKET_LIBRARIES})
+
+    add_library(socket.serial
+        src/buffer.c
+        src/compat.c
+        src/auxiliar.c
+        src/options.c
+        src/timeout.c
+        src/io.c
+        src/usocket.c
+        src/serial.c)
+    set_target_properties(socket.serial PROPERTIES
+        RUNTIME_OUTPUT_DIRECTORY "socket"
+        LIBRARY_OUTPUT_DIRECTORY "socket"
+        PREFIX ""
+        RUNTIME_OUTPUT_NAME "socket"
+        LIBRARY_OUTPUT_NAME "socket")
```

Just by analogy:
```suggestion
        RUNTIME_OUTPUT_NAME "serial"
        LIBRARY_OUTPUT_NAME "serial")
```
???